target_postgres/json_schema.py

```diff
@@ -189,15 +200,43 @@ def make_nullable(schema):
     `is_nullable` will return true on the output.
     :return: dict, JSON Schema
     """
-    type = get_type(schema)
-    if NULL in type:
+    t = get_type(schema)
+    if NULL in t:
         return schema
 
     ret_schema = deepcopy(schema)
-    ret_schema['type'] = type + [NULL]
+    ret_schema['type'] = t + [NULL]
     return ret_schema
 
 
+class Cachable(dict):
+    '''
+    The simplified json_schemas we produce are idempotent. ie, if you simplify a simplified
+    json_schema, it will return the same thing. We wrap the `dict` object with a few
+    helpers which extend it so that we avoid recursion in some instances.
+    '''
+    def __init__(self, raw_dict, simplified=True):
+        self._c = None
+        super(Cachable, self).__init__(self, **raw_dict)
+
+    def __hash__(self):
+        return self._comparator().__hash__()
+
+    def deepcopy(self):
+        s = deepcopy(self)
+        s._c = self._c
+        return s
+
+    def _comparator(self):
+        if not self._c:
+            self._c = json.dumps(self, sort_keys=True)
```

Comment:
Any performance hit here?

Reply:
Yeah, but just once and only on schema messages (at least should be). The reason for having this is so that this doesn't get run all the time.